internal/core/cppapi_data.cpp

```diff
@@ -193,8 +193,18 @@ void fence(
         auto coordinate = transform_coordinate(x, y);
 
         auto validate_boundary = [&] (const int voxel, Axis const& axis) {
-            if(!axis.inrange(coordinate[voxel]))
-                noval_indicies.push_back(i * nsamples);
+            if (!axis.inrange(coordinate[voxel])) {
+                if (fillValue != nullptr) {
+                    noval_indicies.push_back(i * nsamples);
+                } else {
+                    const std::string coordinate_str =
+                        "(" +std::to_string(x) + "," + std::to_string(y) + ")";
+                    throw std::runtime_error(
```

Comment:
I kept it the same as it was like this earlier. Yes, I agree we should return bad requests in this case, but the question is should we add that in the Golang layer or here?

Reply:
If you throw detail::bad_request here, the correct error type will be propagated correctly all the way back to the user. I.e. detail::bad_request ultimately ends up as a HttpBadRequest

Reply:
In theory it would have been better to make this change in its own commit, but I can accept it being together as it is relevant to the changed code...